### PR TITLE
Fix Full Disk Access inaccuracies.

### PR DESCRIPTION
**Pull request by DonNateR**
Fixes issue #334

### CODE REVIEW

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 1]
```diff
@@ -17,9 +17,15 @@ image::images/select-fda.png[Select Full Disk Access]
...
 4. In the lower-left corner of the panel, click the **Lock button** and enter your username and password. You can now add the `elastic-agent` or `esensor` file.
 
-5. Click the + button to view Finder. Navigate to the `/Library/Endgame` directory, select the `elastic-agent` or `esensor` file, and then click *Open*.
+5. Click the + button to view Finder. Select the file that pertains most to your Endpoint configuration: 
++
+- Endpoint Security: Navigate to the directory where you downloaded your `elastic-agent` file and select that file.
++
+- Elastic Endgame: Naviagate to `/Library/Endgame` and select the `esensor` file.
+
+6. After you're selected the applicable file, click **Open**. This gives Elastic Security full disk access.
```
Comment (jmikell821) [inline docs/getting-started/sensor-full-disk-access.asciidoc:26]: change `you're` --> `you've`

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 2]
```diff
@@ -17,9 +17,15 @@ image::images/select-fda.png[Select Full Disk Access]
 +
 4. In the lower-left corner of the panel, click the **Lock button** and enter your username and password. You can now add the `elastic-agent` or `esensor` file.
 
-5. Click the + button to view Finder. Navigate to the `/Library/Endgame` directory, select the `elastic-agent` or `esensor` file, and then click *Open*.
+5. Click the + button to view Finder. Select the file that pertains most to your Endpoint configuration: 
++
+- Endpoint Security: Navigate to the directory where you downloaded your `elastic-agent` file and select that file.
++
+- Elastic Endgame: Naviagate to `/Library/Endgame` and select the `esensor` file.
```
Comment (jmikell821) [inline docs/getting-started/sensor-full-disk-access.asciidoc:24]: spelling - change `Naviagate` --> `Navigate`

**Comment by DonNateR** [comment]
Preview link: https://security-docs_335.docs-preview.app.elstc.co/guide/en/security/master/sensor-full-disk-access.html

**Comment by DonNateR** [comment]
New [Preview](https://security-docs_335.docs-preview.app.elstc.co/guide/en/security/master/sensor-full-disk-access.html)

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 3]
```diff
@@ -3,23 +3,92 @@
...
+[[macos-permissions]]
+== macOS permissions
+
+The behavior of the Endgame Sensor and Elastic Endpoint differs based on your macOS version. MDM/JAMF users can pre-approve all full disk access without granting permission to the sensors. However, non-MDM/JAMF users will be prompted or not prompted to enable full disk access for required security files depending on the macOS version and sensor type:
+
+**Endgame Sensor**
+
+- `10.13, 10.14, 10.15`: Users cannot proceed with installation without first granting the sensor the ability to load a <<enable-kernel-extension,kernel extension>>. During installation, you will be prompted to go to **System preferences** and approve loading the kernel. Upon approval, installation proceeds. 
+
+- `11.0` (Big Sur): Users cannot proceed with installation without first granting the sensor the ability to load <<system-extension>>. During installation, you will be prompted to go to **System preferences** and approve loading the system extension. Upon approval, installation proceeds.
```
Comment (ferullo) [inline docs/getting-started/sensor-full-disk-access.asciidoc:18]:
```suggestion
- `11.0` (Big Sur): Users cannot proceed with installation without first granting the sensor the ability to load <<system-extension>>. During installation, you will be prompted to go to **System preferences** and approve loading the system extension. After approving the system extension a second pop up will appear to approve Network Filtering that needs to be approved. After both approvals, installation proceeds. 
```

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 4]
```diff
@@ -3,23 +3,92 @@
...
 3. In the left pane, select **Full Disk Access**.
 +
 --
 image::images/select-fda.png[Select Full Disk Access]
 --
 +
-4. In the lower-left corner of the panel, click the **Lock button** and enter your username and password. You can now add the `elastic-agent` or `esensor` file.
+4. In the lower-left corner of the panel, click the **Lock button** and enter your username and password. 
++
+5. Click the + button to view Finder. Find the system extension `com.endgame.systemextension` and select the system extension.
```
Comment (ferullo) [inline docs/getting-started/sensor-full-disk-access.asciidoc:59]:
```suggestion
5. Click the + button to view Finder. Find the system extension `com.endgame.systemextension` (Endgame Sensor) or `co.elastic.systemextension` (Elastic Endpoint) and select the system extension.
```

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 5]
```diff
@@ -3,23 +3,92 @@
...
+- `11.0` (Big Sur): Users cannot proceed with installation without first granting the sensor the ability to load <<system-extension>>. During installation, you will be prompted to go to **System preferences** and approve loading the system extension. Upon approval, installation proceeds. 
++
+You also must grant full disk access to `com.endgame.systemextension`
+
+- `10.14.6+, 10.15, 11.0`: Grant the <<endpoint-endgame-sensor,esensor>> full disk access.
+
+
+**Elastic Endpoint**
+
+No prompts to approve the <<enable-kernel-extension,kernel extension>>, <<system-extension>>, or the <<endpoint-endgame-sensor,elastic-endgame>>, due to installation happening through the {agent}. After installation, Endpoint policies **will fail** to detect events until you approve and enable full disk access for each version:
```
Comment (ferullo) [inline docs/getting-started/sensor-full-disk-access.asciidoc:27]: Perhaps also mention that until the user approves the things needed that Policy will not be in a success state in Kibana, so the Kibana user can see what macOS hosts are not fully protected.

**Comment by DonNateR** [comment]
@ferullo: Added the popup step, let me know if there's anything else I need to add before merging.

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 6]
```diff
@@ -3,23 +3,92 @@
...
+- `10.14.6+, 10.15, 11.0`: Grant the <<endpoint-endgame-sensor,esensor>> full disk access.
+
+
+**Elastic Endpoint**
+
+No prompts to approve the <<enable-kernel-extension,kernel extension>>, <<system-extension>>, or the <<endpoint-endgame-sensor,elastic-endgame>>, due to installation happening through the {agent}. After installation, Endpoint policies **will fail** to detect events until you approve and enable full disk access for each version, as reflected in the **Administartion > Endpoints** page of the security application.
+
+- `10.13, 10.14, 10.15`: <<enable-kernel-extension,kernel extension>>.
+
+- `11.0` (Big Sur): <<system-extension>>, as well as `co.endpoint.systemextension`.
```
Comment (ferullo) [inline docs/getting-started/sensor-full-disk-access.asciidoc:31]:
```suggestion
- `11.0` (Big Sur): <<system-extension>>, as well as `co.elastic.systemextension`.
```

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 7]
```diff
@@ -3,23 +3,92 @@
...
+- `11.0` (Big Sur): Users cannot proceed with installation without first granting the sensor the ability to load <<system-extension>>. During installation, you will be prompted to go to **System preferences** and approve loading the system extension. Upon approval, a second prompt appears to enable **Network Filtering**. Approve this final prompt for installation to proceed.
++
+You also must grant full disk access to `com.endgame.systemextension`
+
+- `10.14.6+, 10.15, 11.0`: Grant the <<endpoint-endgame-sensor,esensor>> full disk access.
+
+
+**Elastic Endpoint**
+
+No prompts to approve the <<enable-kernel-extension,kernel extension>>, <<system-extension>>, or the <<endpoint-endgame-sensor,elastic-endgame>>, due to installation happening through the {agent}. After installation, Endpoint policies **will fail** to detect events until you approve and enable full disk access for each version, as reflected in the **Administartion > Endpoints** page of the security application.
```
Comment (ferullo) [inline docs/getting-started/sensor-full-disk-access.asciidoc:27]:
```suggestion
No prompts to approve the <<enable-kernel-extension,kernel extension>>, <<system-extension>>, or the <<endpoint-endgame-sensor,elastic-endgame>>, due to installation happening through the {agent}. After installation, Endpoint policies **will fail** to detect events until you approve and enable full disk access for each version, as reflected in the **Administration > Endpoints** page of the security application.
```

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 8]
```diff
@@ -3,23 +3,91 @@
...
+- `11.0` (Big Sur): Users cannot proceed with installation without first granting the sensor the ability to load <<system-extension>>. During installation, you will be prompted to go to **System preferences** and approve loading the system extension. Upon approval, a second prompt appears to enable **Network Filtering**. Approve this final prompt for installation to proceed.
++
+You also must grant full disk access to `com.endgame.systemextension`
+
+- `10.14.6+, 10.15, 11.0`: Grant the <<endpoint-endgame-sensor,esensor>> full disk access.
+
+
+**Elastic Endpoint**
+
+No prompts to approve the <<enable-kernel-extension,kernel extension>>, <<system-extension>>, or the <<endpoint-endgame-sensor,elastic-endgame>>, due to installation happening through the {agent}. After installation, Endpoint policies **will fail** to detect events until you approve and enable kernel or system extension loading and full disk access for each version, as reflected in the **Administration > Endpoints** page of the security application.
```
Comment (jmikell821) [inline docs/getting-started/sensor-full-disk-access.asciidoc:27]: I feel like these bullet points can be a little confusing w/ just the links in them. What about adding `"Approve the kernel extension", etc. as the link text?

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 9]
```diff
@@ -3,23 +3,91 @@
...
 Elastic Endpoint Security requires full disk access to protect you from malware and other cybersecurity threats. Full Disk Access permissions is a new privacy feature introduced in macOS Mojave (10.14) that prevents some applications from accessing your data. This means you need to manually grant permission for Elastic Endpoint Security to access these protected areas of your Mac.
 
-This article describes how to enable full disk access for both the Elastic Agent, required in order to enable Elastic Endpoint Security, and the legacy Endgame sensor.
+This article describes how to enable full disk access for the required security system extensions, the Elastic Endpoint sensor, and the legacy Endgame sensor.
+
+[discrete]
+[[macos-permissions]]
+== macOS permissions
+
+The behavior of the Endgame Sensor and Elastic Endpoint differs based on your macOS version. MDM/JAMF users can pre-approve all full disk access without granting permission to the sensors. However, non-MDM/JAMF users will be prompted or not prompted to enable full disk access for required security files depending on the macOS version and sensor type:
```
Comment (jmikell821) [inline docs/getting-started/sensor-full-disk-access.asciidoc:12]: You flip between capitalizing the Endgame `sensor` and not. `Endgame sensor` is fine.

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 10]
```diff
@@ -3,23 +3,91 @@
...
+
+**Endgame Sensor**
+
+- `10.13, 10.14, 10.15`: Users cannot proceed with installation without first granting the sensor the ability to load a <<enable-kernel-extension,kernel extension>>. During installation, you will be prompted to go to **System preferences** and approve loading the kernel. Upon approval, installation proceeds. 
+
+- `11.0` (Big Sur): Users cannot proceed with installation without first granting the sensor the ability to load <<system-extension>>. During installation, you will be prompted to go to **System preferences** and approve loading the system extension. Upon approval, a second prompt appears to enable **Network Filtering**. Approve this final prompt for installation to proceed.
++
+You also must grant full disk access to `com.endgame.systemextension`
```
Comment (jmikell821) [inline docs/getting-started/sensor-full-disk-access.asciidoc:20]:
```suggestion
You also must grant full disk access to `com.endgame.systemextension`.
```

Comment (jmikell821) [inline docs/getting-started/sensor-full-disk-access.asciidoc:20]: I would also put this on the previous line.

File: docs/getting-started/sensor-full-disk-access.asciidoc (hunk 9 above)
Comment (jmikell821) [inline docs/getting-started/sensor-full-disk-access.asciidoc:12]:
```suggestion
The behavior of the Endgame Sensor and Elastic Endpoint differs based on your macOS version. MDM/JAMF users can pre-approve all full disk access without granting permission to the sensors. However, depending on the macOS version and sensor type, non-MDM/JAMF users may be prompted to enable full disk access for required security files.
```

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 11]
```diff
@@ -3,23 +3,91 @@
...
+[[system-extension]]
+== System extension
+
+To fully protect endpoints from malware and other cybersecurity threats when using Elastic Endpoint with system extensions, full-disk access must be enabled for the system extension during <<install-endpoint,installation>> on macOS Big Sur (11.0) and later.  
+
+--
+image::images/system-extension-prompt.png[System Extension Prompt]
+--
+
+If you select **Ok** and continue installation, you'll still need to enable Full Disk Access for the system extension.
```
Comment (jmikell821) [inline docs/getting-started/sensor-full-disk-access.asciidoc:45]:
```suggestion
If you select **OK** and continue installation, you'll still need to enable Full Disk Access for the system extension.
```

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 12]
```diff
@@ -3,23 +3,91 @@
...
+
+- `11.0` (Big Sur): <<system-extension>>, as well as `co.elastic.systemextension`.
+
+- - `10.14.6+, 10.15, 11.0`: <<endpoint-endgame-sensor,elastic-endgame>>
+
+[discrete]
+[[system-extension]]
+== System extension
+
+To fully protect endpoints from malware and other cybersecurity threats when using Elastic Endpoint with system extensions, full-disk access must be enabled for the system extension during <<install-endpoint,installation>> on macOS Big Sur (11.0) and later.
```
Comment (jmikell821) [inline docs/getting-started/sensor-full-disk-access.asciidoc:39]:
```suggestion
To fully protect endpoints from malware and other cybersecurity threats when using Elastic Endpoint with system extensions, Full Disk Access must be enabled for the system extension during <<install-endpoint,installation>> on macOS Big Sur (11.0) and later.  
```

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 13]
```diff
@@ -3,23 +3,91 @@
...
 --
 image::images/select-fda.png[Select Full Disk Access]
 --
 +
-4. In the lower-left corner of the panel, click the **Lock button** and enter your username and password. You can now add the `elastic-agent` or `esensor` file.
+4. In the lower-left corner of the panel, click the **Lock button** and enter your username and password. 
++
+5. Click the + button to view Finder. Find the system extension `com.endgame.systemextension` (Endgame Sensor) or `co.elastic.systemextension` and select. 
+
+The system extension now has full disk access. However, for both the {agent} and Elastic Endgame sector to detect events from a macOS host, you must enable full disk access for the file most relevant to your security set up.
```
Comment (jmikell821) [inline docs/getting-started/sensor-full-disk-access.asciidoc:61]:
```suggestion
The system extension now has full disk access. However, for both the {agent} and Elastic Endgame sensor to detect events from a macOS host, you must enable full disk access for the file most relevant to your security setup.
```

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 14]
```diff
@@ -3,23 +3,91 @@
...
+== Elastic Endpoint and Endgame sensor
+
+The `elastic-endpoint` files appear after you've downloaded and installed the {agent} with <<install-endpoint,Endpoint Security Integration>>. Similarly, the `esensor` file for Elastic Endgame appears once you've downloaded the sensor on your host. 
+
+
+1. Open the **System Preferences** application.
++
+2. Click **Security and Privacy**. On the Security and Privacy panel, select the **Privacy** tab.
++
+3. In the left pane, select **Full Disk Access**. Find the system extension `com.endgame.systemextension`
```
Comment (jmikell821) [inline docs/getting-started/sensor-full-disk-access.asciidoc]:
```suggestion
3. In the left pane, select **Full Disk Access**. Find the system extension `com.endgame.systemextension`. 
```

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 15]
```diff
@@ -3,23 +3,91 @@
...
 
-5. Click the + button to view Finder. Navigate to the `/Library/Endgame` directory, select the `elastic-agent` or `esensor` file, and then click *Open*.
+6. After you've selected the applicable file, click **Open**. 
 
-6. In the **Privacy** tab, confirm that the `elastic-agent` or `esensor` file appears in the list of applications that have full access permission, as seen in the following image:
+7. In the **Privacy** tab, confirm that the `elastic-agent` or `esensor` file appears in the list of applications that have full access permission.
 
 
-Elastic Endpoint Security now has the access required to fully protect your system.
+Elastic Endpoint now has the access required to protect your system thoroughly.
```
Comment (jmikell821) [inline docs/getting-started/sensor-full-disk-access.asciidoc:93]: I like what you had previously. 

```suggestion
Elastic Endpoint now has the access required to fully protect your system.
```

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 16]
```diff
@@ -3,23 +3,91 @@
...
+- - `10.14.6+, 10.15, 11.0`: <<endpoint-endgame-sensor,elastic-endgame>>
+
+[discrete]
+[[system-extension]]
+== System extension
+
+To fully protect endpoints from malware and other cybersecurity threats when using Elastic Endpoint with system extensions, full-disk access must be enabled for the system extension during <<install-endpoint,installation>> on macOS Big Sur (11.0) and later.  
+
+--
+image::images/system-extension-prompt.png[System Extension Prompt]
```
Comment (ferullo) [inline docs/getting-started/sensor-full-disk-access.asciidoc:42]: This isn't the image we should show. It's output from Endgame Sensors installer but isn't relevant to Elastic Endpoint and it doesn't really show the user the screen they will see that they need to take action on in System Preferences.

It's probably fine to show just one of the below screens and tell them to approve whichever of "Endgame, Inc", "Endgame", or "ElasticEndpoint" is displayed.

**On macOS 10.15 and earlier, for Endgame Sensor and Elastic Endpoint they'll see this screen**
image

**On macOS 11 for Endgame Sensor they'll see this screen**
image

**On macOS 11 for Elastic Endpoint they'll see this screen**
image

**Review by jmikell821 — CHANGES_REQUESTED** [review]
@DonNateR I added a few comments. Not a huge deal, but I think we should cap `Full Disk Access.` In the articles I've read, it typically is.

File: docs/getting-started/sensor-full-disk-access.asciidoc (hunk 11 above)
Comment (ferullo) [inline docs/getting-started/sensor-full-disk-access.asciidoc:45]:
```suggestion
After you select **Ok** on macOS 11 you'll get a prompt to approve network content filtering, click Allow. Then enable Full Disk Access for the system extension.
```

Depending on the product the wording will vary slightly.

The Network Filtering pop up will look like this for Elastic Endpoint
image


For Endgame Sensor it'll look like this
image

File: docs/getting-started/sensor-full-disk-access.asciidoc (hunk 14 above)
Comment (DonNateR) [reply docs/getting-started/sensor-full-disk-access.asciidoc]: We actually don't need the system extension step in this section, so I removed it.

**Review by ferullo — APPROVED** [review]

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 17]
```diff
@@ -1,25 +1,93 @@
...
+
+**Elastic Endpoint**
+
+No prompts appear to approve the <<enable-kernel-extension,kernel>>, <<system-extension>>, or the <<endpoint-endgame-sensor,elastic-endgame>>, due to installation happening through the {agent}. After installation, Endpoint policies **will fail** to detect events until you approve and enable kernel or system extension loading and Full Disk Access for each version, as reflected in the **Administration > Endpoints** page of the security application.
+
+- `10.13, 10.14, 10.15`: Approve the <<enable-kernel-extension,kernel>>.
+
+- `11.0` (Big Sur): Give Full Disk Access to the <<system-extension>>, as well as `co.elastic.systemextension`.
+
+- - `10.14.6+, 10.15, 11.0`: Grant the <<endpoint-endgame-sensor,elastic-endgame>> Full Disk Access.
```
Comment (jmikell821) [inline docs/getting-started/sensor-full-disk-access.asciidoc:33]:
```suggestion
- `10.14.6+, 10.15, 11.0`: Grant the <<endpoint-endgame-sensor,elastic-endgame>> Full Disk Access.
```
Why are we calling this elastic-endgame? They are either giving FDA to the `elastic-endpoint` or `esensor` file right?

File: docs/getting-started/sensor-full-disk-access.asciidoc [hunk 18]
```diff
@@ -1,25 +1,93 @@
...
+You also must grant Full Disk Access to `com.endgame.systemextension`.
+
+- `10.14.6+, 10.15, 11.0`: Grant the <<endpoint-endgame-sensor,esensor>> Full Disk Access.
+
+
+**Elastic Endpoint**
+
+No prompts appear to approve the <<enable-kernel-extension,kernel>>, <<system-extension>>, or the <<endpoint-endgame-sensor,elastic-endgame>>, due to installation happening through the {agent}. After installation, Endpoint policies **will fail** to detect events until you approve and enable kernel or system extension loading and Full Disk Access for each version, as reflected in the **Administration > Endpoints** page of the security application.
+
+- `10.13, 10.14, 10.15`: Approve the <<enable-kernel-extension,kernel>>.
```
Comment (jmikell821) [inline docs/getting-started/sensor-full-disk-access.asciidoc:29]:
```suggestion
- `10.13, 10.14, 10.15`: Approve the <<enable-kernel-extension,kernel extension>>.
```

**Pull request merged**